### ISSUE TITLE
Extension to 16 Audio Channels

### ISSUE THREAD

**Issue by EkimRemarc** [moritzvieli/rocketshow#109]
RocktShow seems to have the right DNA compared to other multitrackers I have been used since years. 

For Me RocketShow needs two extensions to get more value:

First: I would like to use 16 Audio Channels which enables to have backings for each musician + orchestra in stereo plus a master click & band click. (This is not working in 2.3.5 tested with Behringer X32)

Second: I would suggest to implement a softswitch which globaly enables / disables audio busses. Reason is when a real musician is missing I must switch on his tracks for all compositions for a complete GIG just one time instead of modifying all compositions.

**Comment by DavidOpgh** [comment]
Hello! RS does have a lot of nice features compared to other multirackers I've tried.

1) I've only used a stereo channel audio with RS. What happens when you try to connect it to the X32?

2) I would think this would easier to implement directly from from X32 than from RS.

**Comment by moritzvieli** [comment]
@EkimRemarc Regarding Behringer X32: Same question from my side. We also use a Behringer X32 with 3-4 channels and this works. Does it only work until a specific number of channels? Did you check the 8-8 routing table?

Regarding the 2nd requirement: We're already discussing similar features here: https://github.com/moritzvieli/rocketshow/issues/96

As @DavidOpgh already mentioned: I'd also suggest to map all channels and just mute them in the X32 as required. This is also, what we do live: For 1-2 special songs with guest musicians, we just added separate backing tracks in a channel and mute them, when they're live with us on stage.

**Comment by EkimRemarc** [comment]
It is working up to a certain amount of channels (8). You can configure more but they will not routed to the Behringer and RocketShow shows up a failure message starting while configuring the 9th channel. Up to 8 channels it works fine.

**Comment by moritzvieli** [comment]
@EkimRemarc Can you show your routing configuration in the Behringer X32? Can you post the errormessage here (screenshot or download the logs from the Rocket Show settings)?

**Comment by DavidOpgh** [comment]
@EkimRemarc did you get this issue resolved? Behringer had a sale and I bought an XR18 to test Rocket Show with more than 8 channels.

**Comment by DavidOpgh** [comment]
I've started looking at this issue and I cam confirm RS has an issue with trying to assign more than 8 channels from my xr18

![20240830_174830.jpg](https://github.com/user-attachments/assets/1785dbcf-3445-437e-ad0f-aefa8fd4ef75)

![20240830_175057.jpg](https://github.com/user-attachments/assets/aad85e36-fafd-43af-adf5-6a02516549f0)

![20240830_175141.jpg](https://github.com/user-attachments/assets/ba89cd00-f682-4176-abcc-bb02ca489cb4)

![20240830_175242.jpg](https://github.com/user-attachments/assets/08f39853-14ea-4bbb-a18f-6b81dac5f5aa)

**Comment by moritzvieli** [comment]
@DavidOpgh Just to be sure: I know, you route audio from the interface in the X32 in pairs of 8 channels. Did you set up routing in the X32 to more than 8 channels?

E.g. something like this?
<img width="1011" alt="image" src="https://github.com/user-attachments/assets/51daa7f4-4899-4163-9af7-d8808a8f37ae">

**Comment by DavidOpgh** [comment]
![20240902_080051.jpg](https://github.com/user-attachments/assets/e3fd48a9-fcf4-4226-aa8c-29e38f79cbe1)

![20240902_080939.jpg](https://github.com/user-attachments/assets/266d9637-9f2f-4cf7-8f44-8e10b291ffae)

![20240902_081015.jpg](https://github.com/user-attachments/assets/675091c7-4d5b-4a75-974c-51eaa568c21f)

![20240902_081032.jpg](https://github.com/user-attachments/assets/2d49ec22-bbdb-4c1f-bc43-fc9bba0d76e4)

![20240902_081046.jpg](https://github.com/user-attachments/assets/0f05338f-545e-4375-898c-12a55deed3fd)

![20240902_081058.jpg](https://github.com/user-attachments/assets/be0c4acb-4a65-4576-85fd-7ad2d3ad0368)

**Comment by DavidOpgh** [comment]
![20240902_081032.jpg](https://github.com/user-attachments/assets/6a462c97-2501-489a-8a02-1ca6aea601f8)

![20240902_081046.jpg](https://github.com/user-attachments/assets/02cdd6fa-166a-41b1-b921-db9dd609cad1)

![20240902_081058.jpg](https://github.com/user-attachments/assets/c418d36b-6662-43c7-a14e-64113a5c4d76)

**Comment by DavidOpgh** [comment]
@moritzvieli have you been able to reproduce this error on your end?

**Comment by moritzvieli** [comment]
I got my hands recently on an X32 rack version and could reproduce the issue. I have not found a solution yet. Not sure, what's causing the problems. I need to check, whether Gstreamer (Rocket Show's playback backend) is an issue or the X32 on Linux maybe.

**Comment by DavidOpgh** [comment]
Thanks for the update.

**Comment by DavidOpgh** [comment]
@moritzvieli 
I did a quick Google search. Did you see this? It seems to be related if not exactly the same. 
https://stackoverflow.com/questions/61728296/gstreamer-multichannel-greater-than-8-channels

**Comment by DavidOpgh** [comment]
@moritzvieli 
This seems to indicate that there was a hard limit of 8 but was increased to 64 if using a newer version of gstreamer. 
https://stackoverflow.com/questions/68804159/gstreamer-input-from-osxaudiosrc-with-more-than-8-channels

**Comment by moritzvieli** [comment]
> @moritzvieli This seems to indicate that there was a hard limit of 8 but was increased to 64 if using a newer version of gstreamer. https://stackoverflow.com/questions/68804159/gstreamer-input-from-osxaudiosrc-with-more-than-8-channels

This seems to be only related to OSX. I don't think, this fix should change something on the Raspberry Pi. Also, it has been released in FEB 2022. It should have been included on my test raspi V4, which has an image built in JUN 2023.

**Comment by moritzvieli** [comment]
> @moritzvieli This seems to indicate that there was a hard limit of 8 but was increased to 64 if using a newer version of gstreamer. https://stackoverflow.com/questions/68804159/gstreamer-input-from-osxaudiosrc-with-more-than-8-channels

Yeah, also seen this one. Maybe I need to fiddle around with those caps, although it should not be necessary.

I'll try with a Raspberry Pi 5 or an updated version of Gstreamer. Maybe it's already fixed.

**Comment by DavidOpgh** [comment]
> > @moritzvieli This seems to indicate that there was a hard limit of 8 but was increased to 64 if using a newer version of gstreamer. https://stackoverflow.com/questions/68804159/gstreamer-input-from-osxaudiosrc-with-more-than-8-channels
> 
> Yeah, also seen this one. Maybe I need to fiddle around with those caps, although it should not be necessary.
> 
> I'll try with a Raspberry Pi 5 or an updated version of Gstreamer. Maybe it's already fixed.

Thanks for the update. 64 might be excessive although I can see using more than 8 tracks of audio

**Comment by DavidOpgh** [comment]
> > [@moritzvieli](https://github.com/moritzvieli) This seems to indicate that there was a hard limit of 8 but was increased to 64 if using a newer version of gstreamer. https://stackoverflow.com/questions/68804159/gstreamer-input-from-osxaudiosrc-with-more-than-8-channels
> 
> Yeah, also seen this one. Maybe I need to fiddle around with those caps, although it should not be necessary.
> 
> I'll try with a Raspberry Pi 5 or an updated version of Gstreamer. Maybe it's already fixed.

@moritzvieli 
Were you able to test this?

**Comment by moritzvieli** [comment]
@DavidOpgh I was able to test it yesterday evening but unfortunately, I got the same error with the new version. Need to dig further.

**Comment by DavidOpgh** [comment]
> [@DavidOpgh](https://github.com/DavidOpgh) I was able to test it yesterday evening but unfortunately, I got the same error with the new version. Need to dig further.

OK - thanks for the update.
